src/litserve/server.py

```diff
@@ -196,7 +203,11 @@ def __init__(
                 device_list = range(devices)
             self.devices = [self.device_identifiers(accelerator, device) for device in device_list]
 
-        self.workers = self.devices * self.workers_per_device
+        self.inference_workers = self.devices * self.workers_per_device
+        self.preprocess_workers = self.devices * self.num_preprocess_workers
```

Comment:
preprocess workers need to default to non-zero only when the user overrides preprocess, otherwise no preprocess workers should be spawned (and in that case `ready_to_inference_queue` needs to be `None`)